src/scripts/main.js

```diff
@@ -1,3 +1,58 @@
 'use strict';
 
-// write code here
+function rowToObj(row) {
+  const cells = row.querySelectorAll('td');
+
+  return {
+    name: cells[0]?.textContent.trim() || null,
+    position: cells[1]?.textContent.trim() || null,
+    age: Number(cells[2]?.textContent.trim()) || null,
+    salary: parseFloat(cells[3]?.textContent.replace(/[$,]/g, '')) || null,
+  };
+}
+
+function sortTableArr(a, b, col) {
+  if (a[col] < b[col]) {
+    return -1;
+  }
+
+  if (a[col] > b[col]) {
+    return 1;
+  }
+
+  return 0;
+}
+
+function sortTable(element) {
+  if (element.tagName === 'TH') {
+    const col = element.textContent.toLowerCase();
```

Comment:
The `col` variable is derived from the text content of the header, which might not directly match the keys in the objects created by `rowToObj`. Consider mapping the header text to the appropriate object key.